cmd/maya-exporter/app/collector/jiva.go

```diff
@@ -0,0 +1,115 @@
+// Package collector is used to collect metrics by implementing
+// prometheus.Collector interface. See function level comments
+// for more details.
+package collector
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	v1 "github.com/openebs/maya/pkg/apis/openebs.io/stats"
+	"github.com/pkg/errors"
+
+	"github.com/golang/glog"
+)
+
+// jiva implements the Volume interface. It exposes
+// the metrics of a OpenEBS (Jiva) volume.
+type jiva struct {
+	// url is jiva controller's url
+	url string
+	// stats is volume statistics associated with
+	// jiva (cas)
+	stats stats
+}
+
+// Jiva returns jiva's instance
+func Jiva(url *url.URL) *jiva {
+	url.Path = "/v1/stats"
+	return &jiva{
+		url:   url.String(),
+		stats: stats{},
+	}
+}
+
+// getter get stats from jiva controller
+func (j *jiva) get() (v1.VolumeStats, error) {
+	var volStats v1.VolumeStats
+	if err := j.getVolumeStats(&volStats); err != nil {
+		return v1.VolumeStats{}, err
+	}
+	volStats.Got = true
+	return volStats, nil
+}
+
+// getvolumeStats is used to get the response from the Jiva controller
+// which then unmarshalled into the v1.VolumeStats structure.
+func (j *jiva) getVolumeStats(obj *v1.VolumeStats) error {
+	httpClient := http.DefaultClient
+	httpClient.Timeout = 1 * time.Second
+	resp, err := httpClient.Get(j.url)
+
+	if err != nil {
+		return &connErr{
+			errors.Errorf("%v", err),
+		}
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	glog.Info("Got response: ", string(body))
+	err = json.Unmarshal(body, &obj)
+
+	if err != nil {
+		return err
+	}
+
+	resp.Body.Close()
+	return nil
+}
+
+func (j *jiva) parse(volStats v1.VolumeStats) stats {
+	if !volStats.Got {
+		glog.Warningf("%s", "can't parse stats, controller may not be reachable")
+		return stats{}
+	}
+	j.stats.got = true
+	j.stats.casType = "jiva"
+	j.stats.reads, _ = volStats.Reads.Float64()
```

Comment:
Can we catch or handle the error?

Reply:
done, added a method parseFloat64() as well as a new error counter.